### ISSUE TITLE
csm_transaction.log bad JSON in compute_nodes field

### ISSUE THREAD

**Issue by morrone** [IBM/CAST#498]
We still seem to have a lot of bad JSON in the csm_transaction.log.  This is a pretty big issue for us.

Here is a first obvious problem:

We have a transaction of "type":"allocation-step", where the compute_nodes field looks like this:

`"compute_nodes":]`

There is just a closing square brace, but no opening square brace.

This is with csm 1.4.

**Comment by morrone** [comment]
Perhaps not surprisingly, this is also happening in transactions of type "allocation".

**Comment by mew2057** [comment]
What does the compute nodes field look like on the allocation query for an allocation which has this behavior?

**Comment by morrone** [comment]
I just looked at allocation 715416 using csm_allocation_query, which is an allocation with a bad compute_nodes in the csm_transaction.log.  The csm_allocation_query command lists num_nodes as 256, and under compute_nodes there is a list of 256 unique sierra node names.

There are currently 7997 occurances of this in our csm_transaction.log.

This may actually be a longer existing issue, because the first instance of this problem in the csm_transaction log lists the "begin_time" of the allocation as 2018-08-06 18:50:36.227783.
There are 37 instances of this happening thus far today (Jan 10, 2019).

Note that our csm_transaction.log is 36GB in size at this point, and nearly 5.5 million lines.

**Comment by thanh-lam** [comment]
That's helpful to know. Could you also verify that there are other allocations with 256 nodes but didn't have such format problem? I'm thinking maybe there's a limit on how big the compute_nodes field could be. On the clusters I checked, we have only 20-30 nodes max. And, our node names are very short.

**Comment by morrone** [comment]
I did a bit of analysis on a large tail of the csm_transactions.log (my excerpt from the log start on about 2019-12-19, which is significant because it looks like we installed the csm 1.8.0-1756 rpms the day before that).

First, I grep'ed out all of the lines with bad compute_nodes, and then I counted how many of each "num_nodes" size were found.  This is the result:

```
    172 "num_nodes":256
     41 "num_nodes":512
      4 "num_nodes":768
     87 "num_nodes":1024
      6 "num_nodes":1280
      6 "num_nodes":1536
     30 "num_nodes":2048
      4 "num_nodes":2816
      6 "num_nodes":3584
     38 "num_nodes":4096
```

Somewhat interestingly, all of those sizes are a multiple of 256.  But could that be coincidence?

Then I created the list of counts of num_nodes for the same period of time.  There are 261 distinct node counts, so I won't list that here.  But I note that for all of the lines that have a num_nodes that is a multiple of 256, the count matches up exactly with the count of the lines with bad compute_nodes fields.

In other words, all lines that have a num_nodes count that is a multiple of 256 have a bad compute_nodes field.

Note that I checked a couple of the jobs that use a num_nodes of 4296, and those did not have malformed JSON in the compute_nodes field.  On the other hand, the list of compute_nodes is much less than 4296 nodes.  It looks like there are only 200 node names listed for the three 4296 node count jobs that I spot checked.

**Comment by mew2057** [comment]
Looks like the node list resets every 256 nodes. Able to reproduce, working on fix now.

**Comment by mew2057** [comment]
Fix merged
